analyses/org.osate.analysis.binpacking/src/EAnalysis/BinPacking/MacroPreprocessor.java

```diff
@@ -84,10 +85,10 @@ public void loadMacros(File file) {
 	}
 
 	public void substituteMacrosInFile(File file) {
-		try {
+		PrintWriter writer = null;
+		try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
 			File temp = File.createTempFile("TMP", "MPP");
-			PrintWriter writer = new PrintWriter(new FileOutputStream(temp));
-			BufferedReader reader = new BufferedReader(new FileReader(file));
+			writer = new PrintWriter(new FileOutputStream(temp));
```

Comment:
I originally didn't use try-with-resources for `PrintWriter` because of the variable `temp`, and I still can't seem to be able to come up with a way to use it. Do you have any suggestions on how I could do this? Or should I instead close the `BufferedReader` in a the `finally` block so that they're handled the same way?

Reply:
You can do this with two `try` blocks. The whole thing is wrapped in a `try` ... `catch (IOException)` and then have an inner try-with-resources:

```
try {
	File temp = File.createTempFile("TMP", "MPP");
	try (PrintWriter writer = new PrintWriter(new FileOutputStream(temp));
			BufferedReader reader = new BufferedReader(new FileReader(file))) {
		// Do stuff with writer and reader.
	}
	String fname = file.getAbsolutePath();
	file.renameTo(new File("tmp.old"));
	temp.renameTo(new File(fname));
} catch (IOException e) {
	e.printStackTrace();
}
```